storm-core/src/jvm/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -0,0 +1,644 @@
+        
+        List<String> classPathParams = getClassPathParams(stormRoot);
+        List<String> commonParams = getCommonParams();
+        
+        List<String> commandList = new ArrayList<>();
+        //Log Writer Command...
+        commandList.add(javaCmd);
+        commandList.addAll(classPathParams);
+        commandList.addAll(substituteChildopts(_topoConf.get(Config.TOPOLOGY_WORKER_LOGWRITER_CHILDOPTS)));
+        commandList.addAll(commonParams);
+        commandList.add("org.apache.storm.LogWriter"); //The LogWriter in turn launches the actual worker.
+
+        //Worker Command...
+        commandList.add(javaCmd);
+        commandList.add("-server");
+        commandList.addAll(commonParams);
+        commandList.addAll(substituteChildopts(_conf.get(Config.WORKER_CHILDOPTS), memOnheap));
+        commandList.addAll(substituteChildopts(_topoConf.get(Config.TOPOLOGY_WORKER_CHILDOPTS), memOnheap));
+        commandList.addAll(substituteChildopts(OR(
+                _topoConf.get(Config.TOPOLOGY_WORKER_GC_CHILDOPTS),
+                _conf.get(Config.WORKER_GC_CHILDOPTS)), memOnheap));
+        commandList.addAll(getWorkerProfilerChildOpts(memOnheap));
+        commandList.add("-Djava.library.path=" + jlp);
+        commandList.add("-Dstorm.conf.file=" + stormConfFile);
+        commandList.add("-Dstorm.options=" + stormOptions);
+        commandList.add("-Djava.io.tmpdir=" + workerTmpDir);
+        commandList.addAll(classPathParams);
+        commandList.add("org.apache.storm.daemon.worker");
+        commandList.add(_topologyId);
+        commandList.add(_supervisorId);
+        commandList.add(String.valueOf(_port));
+        commandList.add(_workerId);
+        
+        return commandList;
+    }
+
+    @Override
+    public void launch() throws IOException {
+        _type.assertFull();
+        LOG.info("Launching worker with assignment {} for this supervisor {} on port {} with id {}", _assignment,
+                _supervisorId, _port, _workerId);
+        String logPrefix = "Worker Process " + _workerId;
+        ProcessExitCallback processExitCallback = new ProcessExitCallback(logPrefix);
+        _exitedEarly = false;
+        
+        final WorkerResources resources = _assignment.get_resources();
+        final int memOnheap = getMemOnHeap(resources);
+        final String stormRoot = ConfigUtils.supervisorStormDistRoot(_conf, _topologyId);
+        final String jlp = javaLibraryPath(stormRoot, _conf);
+        
+        List<String> commandList = mkLaunchCommand(memOnheap, stormRoot, jlp);
+
+        Map<String, String> topEnvironment = new HashMap<String, String>();
+        @SuppressWarnings("unchecked")
+        Map<String, String> environment = (Map<String, String>) _topoConf.get(Config.TOPOLOGY_ENVIRONMENT);
+        if (environment != null) {
+            topEnvironment.putAll(environment);
+        }
+        topEnvironment.put("LD_LIBRARY_PATH", jlp);
```

Comment:
If I have `JNI.so` and it depends on `someLibrary.so` without `LD_LIBRARY_PATH` `JNI.so` will not be able to find  `someLibrary.so` if both were shipped in the jar together.